processing/src/test/java/io/druid/query/extraction/SubstringDimExtractionFnTest.java

```diff
@@ -0,0 +1,178 @@
+/*
+ * Druid - a distributed column store.
+ * Copyright 2012 - 2015 Metamarkets Group Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.druid.query.extraction;
+import com.google.common.collect.Sets;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import io.druid.jackson.DefaultObjectMapper;
+import org.junit.Assert;
+import org.junit.Test;
+
+import java.util.Arrays;
+import java.util.Set;
+
+/**
+ */
+public class SubstringDimExtractionFnTest
+{
+
+  private static final String[] testStrings = {
+      "apple",
+      "awesome",
+      "asylum",
+      "business",
+      "be",
+      "cool",
+      "a"
+  };
+
+  private static final String[] testPaths = {
+      "/druid/prod/historical",
+      "/druid/prod/broker",
+      "/druid/prod/coordinator",
+      "/druid/demo/historical",
+      "/druid/demo/broker",
+      "/druid/demo/coordinator",
+  };
+
+
+
+  @Test
+  public void testSubstrings()
+  {
+    int index = 1;
+    int len = 3;
+    ExtractionFn extractionFn = new SubstringDimExtractionFn(index, len);
+
+    Set<String> extracted = Sets.newHashSet();
+
+
+    for (String testString : testStrings) {
+      extracted.add(extractionFn.apply(testString));
+    }
+
+    Assert.assertEquals(7, extracted.size());
+    Assert.assertTrue(extracted.contains("ppl"));
```

Comment:
That essentially. Like I mentioned, I wrote my tests for this by copying the regex tester and adapting it to my needs, and it had tests of the form I used. The idea of the bucket test was that several values go in, fewer come out, but I can make them all simple tests when I next have the chance, since it doesn't technically test anything in addition to your above snippet.